xapian-core/tests/api_weight.cc

```diff
@@ -134,6 +134,49 @@ DEFINE_TESTCASE(bm25weight5, backend) {
 }
 
 // Test exception for junk after serialised weight.
+DEFINE_TESTCASE(bm25plusweight3, !backend) {
```

Comment:
Looking at the PR overall, I can't see these other tests in `api_nodb.cc` -- I can see some serialisation checks in the `weight1` testcase, which use variables `bm25plusweight` and `bm25plusweight2`, but these shouldn't conflict with the testcases you're adding here.